client/app/dedicatedCloud/datacenter/datastore/orderUS/dedicatedCloud-datacenter-datastore-orderUS.controller.js

```diff
@@ -0,0 +1,58 @@
+angular.module("App").controller("DedicatedCloudDatacentersDatastoreOrderUSCtrl", function ($scope, $state, $q, OvhHttp, User, serviceName, datacenterId) {
+    "use strict";
+
+    this.selectedOffer = null;
+    this.quantity = 1;
+    this.expressOrderUrl = null;
+
+    this.fetchOffers = () => OvhHttp.get("/order/cartServiceOption/privateCloud/{serviceName}", {
+        rootPath: "apiv6",
+        urlParams: { serviceName }
+    }).then((offers) => _.filter(offers, { family: "datastore" })).then((offers) => {
+        const sortedResult = _.sortBy(offers, (item) => item.prices[0].price.value);
+        this.selectedOffer = _.first(sortedResult);
+        return sortedResult;
+    });
+
+    this.fetchDatagridOffers = () => this.fetchOffers().then((offers) => ({
+        data: offers,
+        meta: {
+            totalCount: offers.length
```

Comment:
You should use lodash _.size() method, to avoid an `Uncaught TypeError: Cannot read property 'length' of undefined` if `offers === undefined`, or you should test `offers` ^^